### DISCUSSION TITLE
Can we make npm link more ergonomic

### DISCUSSION THREAD

**Discussion by jasonwilliams** [npm/feedback#113] [CLI] [answered]
I use both npm link and yarn link, they're great features.
One thing I find frustrating is re-typing out the package name, or having to copy and paste the name from the command line. If you're on the keyboard it breaks your workflow (if copy/paste), or if its a long package name e.g `@babel/plugin-transform-typescript` then it can be cumbersome to type that out.

Is there something we can do similar to `git stash` where you can `pop` the last name you linked?
Or `git checkout -` where i can go back to previous branch without typing it out.

## Workflow of npm link

1. Navigate to `@namespace/super-long-package-name`
2. `yarn link`
3. Navigate back to `my-app`
4. Either type out `yarn link `@namespace/super-long-package-name` or copy and paste command

Step 4 is cumbersome, are there some ideas that make this easier?
Like (obviously these could clash with package names, but to give you an idea):
1. `yarn link -`

**Comment by MylesBorins** [comment]
I guess a challenge I could see with the "pop" model would be that something like that information would be session state, which to the best of my knowledge npm cli doesn't really have.

/cc @npm/cli about how we could go about this.

**Comment by isaacs** [comment, marked as answer]
So, `npm link` _can_ take a folder name (or any other package spec) as an argument.  How about this?

```
npm link ../super-long-package-name
```

or even:

```
cd /long/path/to/namespace/super/long/package/name/
cd ~/projects/my-app
npm link $(cd -; pwd)  #note: your cd settings may require piping that command to &>/dev/null
```

Or, if your shell supports it (most bash versions do):

```
cd /long/path/to/namespace/super/long/package/name/
cd ~/projects/my-app
npm link $OLDPWD
```

Supporting `npm link -` to mean "the last thing globally installed" would require tracking new information about the global install space, which we don't currently do, as @MylesBorins said.  I'm not opposed to adding that, but it's gonna be more work than these other solutions, which would work today.

**Reply by jasonwilliams** [reply]
> So, `npm link` _can_ take a folder name (or any other package spec) as an argument. How about this?
> 
> ```
> npm link ../super-long-package-name
> ```

I think this may be sufficient, thanks @isaacs 
As you said it could add complication to keep state so using the directory should be good for now. (Happy to close this)
